modules/sources/contracts/fetch-tokenpair-data.ts

```diff
@@ -99,11 +103,27 @@ export async function fetchTokenPairData(
         }
     });
 
+    let bptTokenPairs = generateBptTokenPairs(pools);
+    bptTokenPairs.forEach((bptTokenPair) => {
```

Comment:
No need to do it separately, you could check in the `addEffectivePriceCallsToMulticaller`, etc. functions whether token A is the pool address (maybe need to add pool address to the TokenPair type)